utils.py

```diff
@@ -18,3 +32,144 @@ def ensure_data_source_for_derived_data() -> DataSource:
+    usable for that, and also change the URL.
+    """
+    use_test_server = current_app.config.get("ENTSOE_USE_TEST_SERVER", False)
+    if use_test_server:
+        auth_token = current_app.config.get("ENTSOE_AUTH_TOKEN_TEST_SERVER")
+        entsoe.entsoe.URL = "https://iop-transparency.entsoe.eu/api"
+    else:
+        auth_token = current_app.config.get("ENTSOE_AUTH_TOKEN")
+        entsoe.entsoe.URL = "https://transparency.entsoe.eu/api"
+    if not auth_token:
+        click.echo("Setting ENTSOE_AUTH_TOKEN seems empty!")
+        raise click.Abort
+    return auth_token
+
+
+def abort_if_data_empty(data: Union[pd.DataFrame, pd.Series]):
+    if data.empty:
+        click.echo(
+            "Result is empty. Probably ENTSO-E does not provide these forecasts yet ..."
+        )
+        raise click.Abort
+
+
+def parse_from_and_to_dates(from_date: datetime, to_date: datetime, country_timezone: str) -> Tuple[datetime, datetime]:
+    """
+    Parse CLI options (or set defaults)
+    Note:  entsoe-py expects time params as pd.Timestamp
+    """
+    if to_date is None:
+        today_start = datetime.today().replace(
+            hour=0, minute=0, second=0, microsecond=0
+        )
+        to_date = pd.Timestamp(
+            today_start, tzinfo=pytz.timezone(country_timezone)
+        ) + pd.offsets.DateOffset(
+            days=1
+        )  # Add a calendar day instead of just 24 hours, from https://github.com/gweis/isodate/pull/64
+    else:
+        to_date = pd.Timestamp(to_date, tzinfo=pytz.timezone(country_timezone))
+    if from_date is None:
+        from_time = to_date
+    else:
+        from_time = pd.Timestamp(from_date, tzinfo=pytz.timezone(country_timezone))
+    until_time = to_date + pd.offsets.DateOffset(days=1)  # because to_date is inclusive
+    return from_time, until_time
+
+
+def save_entsoe_series(series: pd.Series, sensor: Sensor, entsoe_source: DataSource, country_timezone: str):
+    """
+    Save a series gotten from ENTSO-E to a Flexeasures database.
+    """
+    now = server_now().astimezone(pytz.timezone(country_timezone))
+    belief_times = (
+        (series.index.floor("D") - pd.Timedelta("6H"))
+        .to_frame(name="clipped_belief_times")
+        .clip(upper=now)
+        .set_index("clipped_belief_times")
+        .index
+    )  # published no later than D-1 18:00 Brussels time
+    bdf = BeliefsDataFrame(
+        series,
+        source=entsoe_source,
+        sensor=sensor,
+        belief_time=belief_times,
+    )
+
+    # TODO: evaluate some traits of the data via FlexMeasures, see https://github.com/SeitaBV/flexmeasures-entsoe/issues/3
+    # TODO: deprecate save_to_db (from api.common)
+    if version("flexmeasures") < "0.8":
+        current_app.logger.warning("Calling flexmeasures.api.common.utils.api_utils.save_to_db is deprecated. Consider switching to FlexMeasures >= 0.8.0")
+        deprecated_save_to_db(bdf)
+    else:
+        save_to_db(bdf)
```

Comment:
This function returns a useful return status string. I suggest printing something accordingly. For example, if you run a CLI task twice shortly after one another, the second one might print "beliefs were already saved" in case ``status=="success_but_nothing_new"`.